
#!/usr/bin/env python
PACKAGE = "objectrecognition"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enable", bool_t, 0, "save the crop cloud ", False)
gen.add("quaternion_enable", bool_t, 0, "axis of map orientaion", False)
gen.add("map_enable", bool_t, 0, "Tuning map data", False)

gen.add("scale_x", double_t, 2, "Scale the x of each point", 1.0, -1.0, 1.0)
gen.add("scale_y_min", double_t, 2, "Scale the y of each point", 1.0, -1.0, 1.0)
gen.add("scale_y_max", double_t, 2, "Scale the z of each point", 1.0, -1.0, 1.0)
gen.add("scale_z_min", double_t, 2, "Scale the y of each point", 1.0, -1.0, 1.0)
gen.add("scale_z_max", double_t, 2, "Scale the z of each point", 1.0, -1.0, 1.0)

gen.add("orientation_x_x", double_t, 2, "Quat_x", 1.0, -1.0, 1.0)
gen.add("orientation_y_y", double_t, 2, "Quat_y", 1.0, -1.0, 1.0)
gen.add("orientation_z_z", double_t, 2, "Quat_z", 1.0, -1.0, 1.0)
gen.add("orientation_w_w", double_t, 2, "Quat_w", 1.0, -1.0, 1.0)

gen.add("radius", double_t, 3, "radius size", .005, 0.0, 0.3)
gen.add("resolution", double_t, 3, "resolution size", 0.02, 0.0, 0.3)
gen.add("thereshold",    int_t,    3, "no of points", 1,  1, 10)


exit(gen.generate(PACKAGE, "objectrecognition", "Tuning"))
